### PR TITLE
Settings redesign, ImageVector preview settings

### PR DESCRIPTION
**Pull request by egorikftp**
- Split settings by tabs
- Preparation for https://github.com/ComposeGears/Valkyrie/issues/132

https://github.com/user-attachments/assets/78101733-6750-4672-ac2a-22dabd2fe03d

and updated dialog:
Screenshot 2024-09-04 at 22 35 38

### CODE REVIEW

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/SettingsScreen.kt [hunk 1]
```diff
@@ -93,64 +84,38 @@ private fun SettingsUI(
...
+                    onAction = onAction,
+                    outputFormat = settings.outputFormat,
+                    generatePreview = settings.generatePreview,
+                )
+                2 -> ImageVectorPreviewSettings(
+                    showImageVectorPreview = settings.showImageVectorPreview,
+                    onAction = onAction,
+                )
+                3 -> AboutSettings()
+            }
```
Comment (Goooler) [inline idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/SettingsScreen.kt:118]: I guess these could be placed in an enum or type-safe navigation? e.g

```kt
enum class Screen {
    First {
        override fun Content() {
            TODO("Not yet implemented")
        }
    },
    Second {...}
    Third {...}
    Forth {...}
    ;
    @Composable
    abstract fun Content()
}
```

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/ui/AboutSettings.kt [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import io.github.composegears.valkyrie.ui.foundation.disabled
+import io.github.composegears.valkyrie.ui.foundation.icons.ExternalLink
+import io.github.composegears.valkyrie.ui.foundation.icons.PluginIcon
+import io.github.composegears.valkyrie.ui.foundation.icons.ValkyrieIcons
+import io.github.composegears.valkyrie.ui.foundation.platform.rememberBrowser
+import io.github.composegears.valkyrie.ui.foundation.theme.PreviewTheme
+import io.github.composegears.valkyrie.ui.screen.intro.util.rememberPluginVersion
+
+@Composable
+fun AboutSettings(modifier: Modifier = Modifier) {
```
Comment (Goooler) [inline idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/ui/AboutSettings.kt:39]:
```suggestion
fun AboutSettings(
    modifier: Modifier = Modifier
) {
```

**Review by Goooler — APPROVED** [review]

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/SettingsScreen.kt (hunk 1 above)
Comment (egorikftp) [reply idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/SettingsScreen.kt:118]: can be done with Tiamat and separate tabs will be as a `Screen`.
will take a look 🙂

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/ui/AboutSettings.kt (hunk 2 above)
Comment (egorikftp) [reply idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/ui/AboutSettings.kt]: reworked, completely removed modifier

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/SettingsScreen.kt (hunk 1 above)
Comment (egorikftp) [reply idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/settings/SettingsScreen.kt]: Made all tabs as nested independent screens 🙂

**Review by Goooler — APPROVED** [review]
I love type-safe things.

**Pull request merged**